File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    
    cache = []
    if cacheSize == 0:
        return len(cities) * 5
    
    cities = [city.lower() for city in cities]
    
    for i in cities:
        if i in cache:
            cache.remove(i)
            answer += 1
        else:
            if len(cache) >= cacheSize:
                cache.remove(cache[0])
            answer += 5
        cache.append(i)
            
    
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        List<String> cache = new ArrayList<>();
        int answer = 0;
        
        for (String city : cities) {
            city = city.toLowerCase();
            if (cache.contains(city)) {
                answer++;
                cache.remove(city);
            }
            else {
                answer += 5;
                if (cache.size() == cacheSize && !cache.isEmpty())
                    cache.remove(cache.size() - 1);
            }
            if (cacheSize > 0)
                cache.add(0, city);
        }
        return answer;
    }
}
